nixos/modules/system/boot/luksroot.nix

```diff
@@ -147,6 +147,11 @@ let
            + optionalString dev.bypassWorkqueues " --perf-no_read_workqueue --perf-no_write_workqueue"
            + optionalString (dev.header != null) " --header=${dev.header}";
     cschange = "cryptsetup luksChangeKey ${dev.device} ${optionalString (dev.header != null) "--header=${dev.header}"}";
+    fifo_tmp_fido_pin = "/dev/temporary_fido_pin";
```

Comment:
Can we use mktemp here?

Reply:
Hey SuperSandro, thanks for chipping in.

I just read the documentation on mktemp. It seems a little safer with the fact that it creates a random file filename, so it'll be impossible for another app to know where the pin number is being written to.

However, wouldn't mktemp create a file on-disk and not in-memory?

Reply:
mktemp is mainly used to avoid collisions because you can always use globs to read the file anyway.

> However, wouldn't mktemp create a file on-disk and not in-memory?

Yes it would.

Reply:
How would we get collisions? There's only 1 section in luksroot that reads, makes or deletes a file (block-device) at /dev/temporary_fido_pin. If this is still a problem, maybe we could use a random filename to make a fifo queue in that name, seems like it'll be best of both worlds then.

Reply:
I don't know if it is an issue or not but usually tmp files have no predictable names to account for parallism and other processes.

Reply:
Understandable.

In that case, I'd say the best option would be to randomly name the named pipes similarly to what `mktmp` does. Unless there are any objections from someone, I can adjust the PR to include this!